### DISCUSSION TITLE
Validation trough API

### DISCUSSION THREAD

**Discussion by jasminhusadzic** [schemaorg/schemaorg#3113] [Q&A]
Do you provide api for validation schemas on certain url. Something like you have on UI on https://validator.schema.org/
I have need to integrate schema validator in Cypress tests, but stucked at this point.

Thanks and regards

**Comment by thadguidry** [comment]
Schema.org only provides standards and guidance.  You might ask search engine companies or search on GitHub to see if others have written code in Python, Ruby, Javascript that can validate certain markup and formats like JSON-LD.  The JSON-LD Playground also has code, a mailing list, and a Gitter channel.

**Reply by alexander-schranz** [reply]
I have a similar usecase but searching for something I can run locally. What type of validator is used on validator.schema.org is it on github?

**Reply by thadguidry** [reply]
Here's some docs https://github.com/schemaorg/schemaorg/blob/main/docs/validator.md
@danbri would be able to respond if there are plans to open source the validator or host the code publicly.

**Comment by danbri** [comment]
It is ill-suited for opensource but we should work on interop with similar
opensource tooling.

**Reply by mfhepp** [reply]
A baseline SHACL ruleset would be ideal, IMO - it could be extended by use-case, consumer (Google, Bing, general Semantic Web) and would be minimal and transparent.

It would also be straightforward to build a local validation runtime so that we would avoid the hassle that e.g. the W3C HTML validator is facing wrt to quota management. My guess is that if one provides an API, people will start running automated checks on shopping sites with 50k+ item pages...

**Reply by mfhepp** [reply]
SHACL + a CLI runtime would also be great for CI/CD workflows

**Reply by alexander-schranz** [reply]
From my naive point of view. I think 2 types of tools would be great for CI's not sure if something yet exist. An `extractor` which extracts `microdata`, `rdfa` and `json-ld` from a site and convert all of them into `json-ld` jsons.  

And then a validator here not much would needed if schema.org provides json schemas for all there defined schemas (I could not find any official json schemas yet), as then every json schema validator could be used.

**Reply by rdutta-kareo** [reply]
I want to run a public API for https://validator.schema.org/ on multiple of our Clients websites, to test markup. We want to obtain the results for our React Web project, and display in a Table.

Is there an general API to run this, numerous times?
I am getting 429 Too many requests for calling the API 50 times in an hour. Should I just avoid calling the API entirely, and use the github project locally? At the very least, can you open the website for Iframe, so it can be used in other websites?

**Reply by EvanCarroll** [reply]
Disagree here. Be great to have it open source so we could integrate it into Zola. and CI/CD workflows. Even if it was complex we could A/B against a reimplementation for the purposes of development.
